pvlib/bifacial.py

```diff
@@ -0,0 +1,147 @@
+        DataFrame containing all the detailed outputs and elements calculated
+        for every timestamp of the simulation. Please refer to pvfactors
+        documentation for more details
+
+    References
+    ----------
+    .. [1] Anoma, Marc Abou, et al. "View Factor Model and Validation for
+        Bifacial PV and Diffuse Shade on Single-Axis Trackers." 44th IEEE
+        Photovoltaic Specialist Conference. 2017.
+    """
+
+    # Convert pandas Series inputs to numpy arrays
+    if isinstance(solar_azimuth, pd.Series):
+        solar_azimuth = solar_azimuth.values
+    if isinstance(solar_zenith, pd.Series):
+        solar_zenith = solar_zenith.values
+    if isinstance(surface_azimuth, pd.Series):
+        surface_azimuth = surface_azimuth.values
+    if isinstance(surface_tilt, pd.Series):
+        surface_tilt = surface_tilt.values
+    if isinstance(dni, pd.Series):
+        dni = dni.values
+    if isinstance(dhi, pd.Series):
+        dhi = dhi.values
+
+    # Import pvfactors functions for timeseries calculations.
+    from pvfactors.timeseries import (calculate_radiosities_parallel_perez,
+                                      calculate_radiosities_serially_perez,
+                                      get_average_pvrow_outputs)
+    idx_slice = pd.IndexSlice
+
+    # Build up pv array configuration parameters
+    pvarray_parameters = {
+        'n_pvrows': n_pvrows,
+        'pvrow_height': pvrow_height,
+        'pvrow_width': pvrow_width,
+        'surface_azimuth': surface_azimuth[0],  # not necessary
+        'surface_tilt': surface_tilt[0],        # not necessary
+        'gcr': gcr,
+        'solar_zenith': solar_zenith[0],        # not necessary
+        'solar_azimuth': solar_azimuth[0],      # not necessary
+        'rho_ground': albedo,
+        'rho_front_pvrow': rho_front_pvrow,
+        'rho_back_pvrow': rho_back_pvrow,
+        'horizon_band_angle': horizon_band_angle
+    }
+
+    # Run pvfactors calculations: either in parallel or serially
+    if run_parallel_calculations:
+        df_registries, df_custom_perez = calculate_radiosities_parallel_perez(
+            pvarray_parameters, timestamps, solar_zenith, solar_azimuth,
+            surface_tilt, surface_azimuth, dni, dhi,
+            n_processes=n_workers_for_parallel_calcs)
+    else:
+        inputs = (pvarray_parameters, timestamps, solar_zenith, solar_azimuth,
+                  surface_tilt, surface_azimuth, dni, dhi)
+        df_registries, df_custom_perez = calculate_radiosities_serially_perez(
+            inputs)
+
+    # Get the average surface outputs
+    df_outputs = get_average_pvrow_outputs(df_registries,
+                                           values=['qinc'],
+                                           include_shading=True)
+
+    # Select the calculated outputs from the pvrow to observe
+    ipoa_front = df_outputs.loc[:, idx_slice[index_observed_pvrow,
+                                             'front', 'qinc']].values
```

Comment:
perhaps only a question because I'm not familiar enough with `pvfactors`, but why extract `.values` instead of returning the Series?

Reply:
I just believed that having a 1D numpy array was simpler for these results, but I can adjust it to ``pvlib``'s needs, just let me know!